modules/enum_monitoring/main.py

```diff
@@ -40,6 +40,73 @@
 parser.add_argument('--vpc', required=False, default=False, action='store_true', help='Enumerate VPC flow logs.')
 
 
+def permission_check(pacu, args):
+    """Checks permissions and runs enumeration accordingly"""
+    arguments = [args.cloud_trail, args.cloud_watch, args.shield, args.guard_duty, args.config, args.vpc]
+    enum_all = not any(arguments)
+    confirmed = []
+    if args.cloud_trail or enum_all:
+        try:
+            client = pacu.get_boto3_client('cloudtrail', 'us-east-1')
+            client.describe_trails()
+            confirmed.append('cloudtrail')
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if code != 'AccessDeniedException':
+                pacu.print(code)
+    if args.shield or enum_all:
+        try:
+            client = pacu.get_boto3_client('shield', 'us-east-1')
+            client.get_subscription_state()
+            client.describe_subscription()
+            confirmed.append('shield')
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if code == 'InvalidSignatureException':
+                # TODO Solve BOTO3 Error
+                pass
+            elif code != 'AccessDeniedException':
+                pacu.print(code)
+    if args.guard_duty or enum_all:
+        try:
+            client = pacu.get_boto3_client('guardduty', 'us-east-1')
+            client.list_detectors()
+            confirmed.append('guardduty')
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if code != 'AccessDeniedException':
+                pacu.print(code)
+    if args.config or enum_all:
```

Comment:
This doesn't hit all the API calls that Config makes. It might be a better idea to just wrap the calls in the main function and if there is an error thrown, remove that group from the targets for each for loop iteration